v2/observer/jobs.go

```diff
@@ -203,22 +203,23 @@ func (o *observer) startJobPodTimer(ctx context.Context, pod *corev1.Pod) {
 		pod.Status.Phase != corev1.PodRunning {
 		return
 	}
-	if pod.Annotations[myk8s.AnnotationTimeoutDuration] == "" {
-		return
-	}
 
+	// Attempt to set the timeout per the annotation on the pod itself
 	duration := pod.Annotations[myk8s.AnnotationTimeoutDuration]
 	timeout, err := time.ParseDuration(duration)
-	if err != nil {
+	// Fallback to the max if we are unable to parse timeout value or if it
+	// exceeds the max.
+	if err != nil || timeout > o.config.maxJobLifetime {
 		o.errFn(
-			errors.Wrapf(
-				err,
-				"unable to parse timeout duration %q for pod %q",
+			fmt.Errorf(
+				"unable to parse timeout duration %q for pod %q; "+
+					"using configured maximum of %q",
```

Comment:
Should we use a different message if the issue is that the max timeout allowed by the operator was exceeded? This message will always look like a parsing failure.

Reply:
Fair point.  Was trying to simplify logic but at the expense of clarity in logging, which is never a good idea 😂 

Consolidated logic with this latest iteration... lemme know how it's looking when convenient.